x-pack/plugins/lens/public/datatable_visualization/_visualization.scss

```
@@ -1,3 +1,14 @@
 .lnsDataTable {
   align-self: flex-start;
 }
+
+.lnsDataTable__filter {
+  opacity: 0;
+  transition: opacity 250ms ease-in-out;
+}
+
+.lnsDataTable__cell:hover  {
```

The icons need to reveal on focus as well. 
```suggestion
.lnsDataTable__cell:hover,
.lnsDataTable__cell:focus-within {
```